docs/source/en/model_doc/granite.md

```diff
@@ -57,14 +76,42 @@ input_tokens = tokenizer(prompt, return_tensors="pt")
 output = model.generate(**input_tokens, max_new_tokens=100)
 # decode output tokens into text
 output = tokenizer.batch_decode(output)
-# loop over the batch to print, in this example the batch size is 1
+# loop over the batch to print, in this example, the batch size is 1
 for i in output:
     print(i)
 ```
-
 This model was contributed by [mayank-mishra](https://huggingface.co/mayank-mishra).
+</hfoption>
+</hfoptions>
+
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
+
+The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to int4.
+
+```python
+from transformers import AutoTokenizer, AutoModelForCausalLM, BitsAndBytesConfig
```

Comment:
Wrong model checkpoint here ;)

```suggestion
import torch
from transformers import AutoTokenizer, AutoModelForCausalLM, BitsAndBytesConfig

quantization_config = BitsAndBytesConfig(load_in_4bit=True)
tokenizer = AutoTokenizer.from_pretrained("ibm-granite/granite-3.3-8b-base")
model = AutoModelForCausalLM.from_pretrained("ibm-granite/granite-3.3-8b-base", torch_dtype=torch.bfloat16, device_map="auto", attn_implementation="sdpa", quantization_config=quantization_config)

inputs = tokenizer("Explain quantum computing in simple terms", return_tensors="pt").to("cuda")
outputs = model.generate(**inputs, max_length=50, cache_implementation="static")
print(tokenizer.decode(outputs[0], skip_special_tokens=True))
```

Reply:
Still need to `import torch` and use the larger 8b checkpoint instead of the 2b one.